code/game/machinery/fire_alarm.dm

```diff
@@ -36,31 +36,75 @@ FIRE ALARM
 /obj/machinery/firealarm/update_icon()
 	cut_overlays()
 
+	pixel_x = 0
+	pixel_y = 0
+	if(dir == NORTH)
+		pixel_y = -21
+	else if(dir == SOUTH)
+		pixel_y = 21
+	else if(dir == WEST)
+		pixel_x = -21
+	else if(dir == EAST)
+		pixel_x = 21
```

Comment:
this should really be something triggered by setDir and/or Initialize as with the others

mayhaps we should take advantage of get_base_pixel_x/y procs on /atom/movable or wherever i put them and just call reset_pixel_shifts (put it at /atom or /atom/movable if it isn't already) on initialize/setdir? or don't, i don't really care i can mop it up later if you don't wanna worry about it, just don't have it in update_icon, it's very messy and doesn't need to be called every time.